pkg/util/k8s/client.go

```diff
@@ -125,6 +127,10 @@ func OverrideKubeAPIServer(kubeAPIServerOverride string) {
 		host = hostPort
 		port = "443"
 	}
+	portNum, err := strconv.Atoi(port)
+	if !checks.IsValidPort(portNum) || err != nil {
+		port = "443"
+	}
```

Comment:
here also it seems like silently dropping an error may not be the right thing to do?
maybe the function could return an error?